src/torrent.cpp

```diff
@@ -9586,7 +9558,7 @@ namespace libtorrent
 		// now, iterate over all time critical pieces, in order of importance, and
 		// request them from the peers, in order of responsiveness. i.e. request
 		// the most time critical pieces from the fastest peers.
-		for (std::vector<time_critical_piece>::iterator i = m_time_critical_pieces.begin()
+		for (auto i = m_time_critical_pieces.begin()
```

Comment:
if you would introduce a ``bool first`` variable, I think you could make this a range-for loop too